engines/sci/graphics/celobj32.cpp

```diff
@@ -791,6 +791,43 @@ void CelObj::scaleDrawUncompNoMD(Buffer &target, const Ratio &scaleX, const Rati
 #pragma mark -
 #pragma mark CelObjView
 
+int16 CelObjView::getNumLoops(const GuiResourceId viewId) {
+	Resource *resource = g_sci->getResMan()->findResource(ResourceId(kResourceTypeView, viewId), false);
+
+	if (!resource) {
+		return 0;
+	}
+
+	return resource->data[2];
+}
+
+int16 CelObjView::getNumCels(const GuiResourceId viewId, const int16 loopNo) {
+	Resource *resource = g_sci->getResMan()->findResource(ResourceId(kResourceTypeView, viewId), false);
+
+	if (!resource) {
+		return 0;
+	}
+
+	byte *data = resource->data;
+
+	uint16 loopCount = data[2];
+	if (loopNo >= loopCount) {
+		return 0;
+	}
+
+	const uint16 viewHeaderSize = READ_SCI11ENDIAN_UINT16(data);
+	const uint8 loopHeaderSize = data[12];
+	const uint8 viewHeaderFieldSize = 2;
+
+	byte *loopHeader = data + viewHeaderFieldSize + viewHeaderSize + (loopHeaderSize * loopNo);
+
+	if ((int8)loopHeader[0] != -1) {
+		loopHeader = data + viewHeaderFieldSize + viewHeaderSize + (loopHeaderSize * (int8)loopHeader[0]);
+	}
+
+	return loopHeader[2];
+}
```

Comment:
This is new code, so one could do it properly. It's not like there would be 100 checks anyway and it would be tons of work.

I for example added checks in the new AGI view code, when I rewrote it. I also added tons of checks in the sherlock holmes code that I wrote.

Even in the current SCI view code, there are quite a lot of checks. Quite a few are also missing and that's also bad, but that's not the point. That should also get fixed. Same with the saved game code. My new save game code also checks for all sorts of issues. The current one is prone to crash in all sorts of ways, which is always bad.